### PR TITLE
Add introduction to bit-wise functions and enhance the section

### PR DESCRIPTION
**Pull request by sheepduke**
Change list:

* Added introduction to bit-wise functions.
* Moved URL links to the end of the page.
* Fixed a bug that sub-section immediately after the title will now be parsed properly.
* Merged sub-section `Using complex numbres` to `Complex types`.

### CODE REVIEW

**Comment by sheepduke** [comment]
Summon the fleet. @vindarel

File: numbers.md [hunk 1]
```diff
@@ -477,22 +476,123 @@ bound to a copy of its state before the `let` form.
...
-and return complex numbers when this is the true result. For example:
+Common Lisp also provides many functions to perform bit-wise arithmetic
+operations. Some commonly used ones are listed below, together with their
+C/C++ equivalence.
+
+{:class="table table-bordered table-stripped"}
+| Common  Lisp     | C/C++       | Description                                      |
+|------------------|-------------|--------------------------------------------------|
+| `(logand a b c)` | `a & b & c` | Bit-wise and of multiple operands                |
+| `(logior a b c)` | `a | b | c` | Bit-wise OR of multiple arguments                |
```
Comment (vindarel) [inline numbers.md]: Here "OR" is uppercase when on the other lines the operators are not ("bit-wise and of multiple operands"). Uppercase all operators

**Comment by sheepduke** [comment]
> Why do you move all links to the bottom ? I find it a bit more cumbersome.
> 
> Otherwise, damn good !

It has 2 advantages:
1. It makes each line shorter, i.e. easier to read when manipulating the source.
2. It makes it easier to re-use the same link.

IMHO it acts more like a "references" sub-section that makes the structure more clear...

File: numbers.md (hunk 1 above)
Comment (sheepduke) [reply numbers.md]: Fixed.

**Comment by vindarel** [comment]
Thank you !

**Pull request merged**